### PR TITLE
Enum Impl & Model Cleanup

### PR DESCRIPTION
**Pull request by nateweisz**
**This PR does the following**:
- Replace integer's that represent enum's with the enums themselves
- refractor the model structure to be more similar to the discord docs and endpoints package
- Add missing fields in certain models & document some of what still needs to be done

I went through almost all the models except the guild and message package :)

### CODE REVIEW

File: src/main/java/com/javadiscord/jdi/internal/models/auditlog/AuditLogEntry.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.javadiscord.jdi.internal.models.auditlog;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+// missing auditLogChanges
```
Comment (surajkumar) [inline src/main/java/com/javadiscord/jdi/internal/models/auditlog/AuditLogEntry.java:8]: Can you explain this comments?

File: src/main/java/com/javadiscord/jdi/internal/models/auditlog/AuditLogEntry.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+public record AuditLogEntry(
+    @JsonProperty("target_id") String targetId,
```
Comment (surajkumar) [inline src/main/java/com/javadiscord/jdi/internal/models/auditlog/AuditLogEntry.java:11]: All IDs should be strings, can you confirm and make the changes across the models you have changed? I can do this in a seperate PR too so LMK

**Review by surajkumar — CHANGES_REQUESTED** [review]

File: src/main/java/com/javadiscord/jdi/internal/models/auditlog/AuditLogEntry.java (hunk 2 above)
Comment (nateweisz) [reply src/main/java/com/javadiscord/jdi/internal/models/auditlog/AuditLogEntry.java]: Fixed

File: src/main/java/com/javadiscord/jdi/internal/models/auditlog/AuditLogEntry.java (hunk 1 above)
Comment (nateweisz) [reply src/main/java/com/javadiscord/jdi/internal/models/auditlog/AuditLogEntry.java:8]: We discussed on disc and decided to leave for tech debt

**Comment by nateweisz** [comment]
Closes #65

File: src/main/java/com/javadiscord/jdi/internal/models/guild/WidgetObject.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public record WidgetObject(
+    @JsonProperty("id") long id,
+    @JsonProperty("name") String name,
+    @JsonProperty("instant_invite") String instantInvite,
+    // @JsonProperty("channels") List<PartialChannel> channels, // TODO: future
```
Comment (surajkumar) [inline src/main/java/com/javadiscord/jdi/internal/models/guild/WidgetObject.java:11]: Why are these TODOs and commented out?

Comment (nateweisz) [reply src/main/java/com/javadiscord/jdi/internal/models/guild/WidgetObject.java:11]: It was before we had discussed on discord how to do partial channels, I'll fix that!

**Review by Taz03 — APPROVED** [review]

**Review by surajkumar — APPROVED** [review]

**Pull request merged**